src/sha2.rs

```diff
@@ -1,37 +1,8 @@
 #[cfg(not(openssl))]
-use sha2::{Digest, Sha256 as Ctx};
+pub use sha2::{Digest, Sha256};
 
 #[cfg(openssl)]
-use openssl::sha;
-
-// WINDOWS PART
-
-#[cfg(not(openssl))]
-#[derive(Clone)]
-pub struct Sha256 {
-    ctx: Ctx,
-}
-
-#[cfg(not(openssl))]
-impl Sha256 {
-    pub fn new() -> Sha256 {
-        Sha256 { ctx: Ctx::new() }
-    }
-    pub fn update(&mut self, buf: &[u8]) {
-        self.ctx.update(buf);
-    }
-    pub fn finish(self) -> [u8; 32] {
-        self.ctx.finalize().into()
```

Comment:
this PR is problematic. For example, the `finish()` function used to always return `[u8: 32]`, but now its return type depends on whether you use the openssl or `sha2` implementation. `sha2` returns a `GenericArrat<>` object that can be converted to a slice.